pkg/security/apiserver/registry/podsecuritypolicysubjectreview/rest.go

```diff
@@ -63,23 +62,20 @@ func (r *REST) Create(ctx context.Context, obj runtime.Object, _ rest.ValidateOb
 		return nil, kapierrors.NewInvalid(kapi.Kind("PodSecurityPolicySubjectReview"), "", errs)
 	}
 
+	var err error
+	var matchedConstraints []*securityapi.SecurityContextConstraints
 	userInfo := &user.DefaultInfo{Name: pspsr.Spec.User, Groups: pspsr.Spec.Groups}
-	matchedConstraints, err := r.sccMatcher.FindApplicableSCCs(userInfo, ns)
-	if err != nil {
-		return nil, kapierrors.NewBadRequest(fmt.Sprintf("unable to find SecurityContextConstraints: %v", err))
-	}
-
 	saName := pspsr.Spec.Template.Spec.ServiceAccountName
 	if len(saName) > 0 {
 		saUserInfo := serviceaccount.UserInfo(ns, saName, "")
-		saConstraints, err := r.sccMatcher.FindApplicableSCCs(saUserInfo, ns)
-		if err != nil {
-			return nil, kapierrors.NewBadRequest(fmt.Sprintf("unable to find SecurityContextConstraints: %v", err))
-		}
-		matchedConstraints = append(matchedConstraints, saConstraints...)
+		matchedConstraints, err = r.sccMatcher.FindApplicableSCCs(ns, userInfo, saUserInfo)
+	} else {
+		matchedConstraints, err = r.sccMatcher.FindApplicableSCCs(ns, userInfo)
```

Comment:
Same comment as above (which calls my attention to all the repetition! gahh!)

Reply:
There's some additional copy-pasta code below this. I tried to fight it with generalization, but it fought back with special cases in error handling and logging, so I eventually gave up.